src/wallet/wallet.h

```diff
@@ -961,6 +961,12 @@ class CWallet final : public WalletStorage, private interfaces::Chain::Notificat
 
     std::set<CTxDestination> GetLabelAddresses(const std::string& label) const;
 
+    /**
+     * Mark all outputs in a destination dirty so their cache is
+     * reset and does not return outdated information.
+     */
+    void MarkDestinationsDirty(const std::set<CTxDestination>& destinations);
```

Comment:
Add annotation `EXCLUSIVE_LOCKS_REQUIRED(cs_wallet)` and remove recursive lock in L3149.

Reply:
done